hw1/CBC-HMAC.py

```diff
@@ -51,15 +51,28 @@ def __strip_padding(self, data):
         :param data: input data
         :return: stripped data
         """
-        ?
+        pad_len = data[-1]
+        actual_data, padding = data[:-(pad_len + 1)], data[-(pad_len + 1): -1]
+        for byte in padding: # TODO verify we check all
+            if byte != pad_len:
+                raise "Padding is not valid" #TODO verify what to do in this case
+        expected_pad_len = self.block_len - ((len(actual_data) + 1) % self.block_len) #TODO maybe extract, also used in __pad
```

Comment:
Maybe the check on expected_pad_len is not needed given the previous comment (checking pad_len is not more than block_len and also what we checked about each byte in padding).